processing/src/main/java/org/apache/druid/query/scan/ScanResultValueFramesIterable.java

```diff
@@ -200,26 +229,33 @@ public FrameSignaturePair next()
       // start all the processing
       populateCursor();
       boolean firstRowWritten = false;
-      // While calling populateCursor() repeatedly, currentRowSignature might change. Therefore we store the signature
+      // While calling populateCursor() repeatedly, currentRowSignature might change. Therefore, we store the signature
       // with which we have written the frames
-      final RowSignature writtenSignature = currentRowSignature;
+      final RowSignature writtenSignature = trimmedRowSignature;
```

Comment:
instead of introducing this variable why not use `frameWriterFactory.signature()` ?

Reply:
`frameWriterFactory.signature()` looks cleaner. Thanks for the suggestion, I'll use that instead